### PR TITLE
Migrate codebase to typescript

### PR DESCRIPTION
**Pull request by sync**
This one doesn't have styled component, and wasn't able to update to emotion 10

### CODE REVIEW

File: www/pages/_document.tsx [hunk 1]
```diff
@@ -87,7 +98,11 @@ const clientSideJS = `
...
     document.querySelector('select[name=language]').addEventListener('change', () => document.tune.submit())
     document.querySelector('select[name=time]').addEventListener('change', () => document.tune.submit())
+  })
+`;
 
+const serviceWorkerRegistration = `
+  document.addEventListener('DOMContentLoaded', event => {
     if ('serviceWorker' in navigator) {
       window.addEventListener('load', () => {
         navigator.serviceWorker.register('/_next/static/service-worker.js', { scope: "/" }).then(registration => {
```
Comment (sync) [inline www/pages/_document.tsx:108]: i'm not registering service worker in dev mode to avoid console error about missing file

File: now.json [hunk 2]
```diff
@@ -27,8 +27,8 @@
       "use": "@now/next"
     },
     {
-      "src": "api/index.js",
-      "use": "@now/node-server"
+      "src": "api/index.tsx",
+      "use": "@dblechoc/node-server"
     },
```
Comment (sync) [inline now.json:32]: custom fork of node-server, while waiting for some code to be published, PR just got merged: https://github.com/zeit/now-builders/pull/128#event-2017191650

Comment (hanford) [reply now.json:32]: they're usually really fast about this - might hold off on merging this PR for a few hours to see if they cut another release. (totally cool if we're on `@now/node-server@canary)

**Comment by sync** [comment]
test deploy: https://trends-om4otxb2h.now.sh

File: now.json (hunk 2 above)
Comment (sync) [reply now.json:32]: yes once a new version is published i'll update the PR

**Comment by sync** [comment]
did an audit: still 100%

File: now.json (hunk 2 above)
Comment (hanford) [reply now.json:32]: Looks like they still haven't cut a new release, maybe if we ask them to they will release a new canary?

Super eager to get this PR merged! 🤩

**Comment by hanford** [comment]
@sync also looks like changing languages isn't working 🤔

**Comment by sync** [comment]
i can reproduce it locally, having a look

**Comment by sync** [comment]
@hanford found the problem, was because of using a newer graphql venison on the sever, fixed now

**Comment by sync** [comment]
new url: https://trends-drtphk94t.now.sh

**Comment by hanford** [comment]
@sync awesome, can't wait to get this merged!

**Comment by sync** [comment]
i asked the people at zeit for a new release: https://github.com/zeit/now-builders/pull/128

**Comment by hanford** [comment]
@sync if they don't release a canary today we could merge this as is and either use your fork of the builder (until Zeit releases a new version) or just release this as is and tag `@now/node-server@canary` which theoretically would just start working whenever they do the release.

Thoughts?

**Comment by sync** [comment]
we can merge this as is and as soon as we get a new update from zeit i'll send a new PR

**Comment by sync** [comment]
this repo helped me get started with next.js and now.sh few months ago btw, really liked that I could just check it out and deploy it easily and tweak things

**Comment by sync** [comment]
also if you wan me to change it to `@now/node-server@canary` i'll be happy to do so

**Comment by sync** [comment]
changed to `@now/node-server@canary` it's better

**Comment by hanford** [comment]
🍻🥂Will merge in the morning - thanks again!

**Pull request merged**